descqa/readiness_test.py

```diff
@@ -138,8 +138,11 @@ def _format_row(self, quantity, plot_filename, results):
 
     def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
 
-        all_quantities = sorted(catalog_instance.list_all_quantities(True))
+        #list_of_all_quantities = sorted(map(str,catalog_instance.list_all_quantities(True)))
+        #using True in the previous statement means that the elements require further parsing to remove the catalog name for composite catalogs
+        #all_quantities = [re.sub("'", "", q.split(', ')[-1].split(')')[0]) for q in list_of_all_quantities]
 
+        all_quantities = sorted(catalog_instance.list_all_quantities())
```

Comment:
This change makes the test not backward compatible. There are cases we would want the readiness test to run on all quantities, including native ones. I think the way to go is to add a test option like `exclude_native`, and we can turn it on for composite catalogs.

Reply:
So the commented out lines above (141-143) are what is needed if one sets True in the list_all_quantities call. Perhaps you know a more elegant way to do this but the idea is to parse the string to see if it came from a tuple (hence split on ', '), take the last argument of that split, which is the desired quantity, and remove the trailing bracket and any internal quotes. I have tested that those lines work.  I'll double check, but with those additions the test will continue to be backwards compatible and will work on both native and non-native quantities in composite catalogs.

Reply:
Generally, the test should not assume that the quantity names are strings, nor assume that the quantity names are tuples. The readiness test does assume the quantity names are strings. So what we can do here is (1) add an option to toggle all or just GCR quantities and/or (2) remove any quantities that have non-string names.